config/configtls/configtls.go

```diff
@@ -75,17 +71,12 @@ type Config struct {
 	ReloadInterval time.Duration `mapstructure:"reload_interval"`
 }
 
-// TSLClientSetting contains TLS configurations that are specific to client
-// connections in addition to the common configurations.
-// Deprecated: [v0.96.0] Use ClientConfig instead.
-type TLSClientSetting = ClientConfig
-
 // ClientConfig contains TLS configurations that are specific to client
 // connections in addition to the common configurations. This should be used by
 // components configuring TLS client connections.
 type ClientConfig struct {
 	// squash ensures fields are correctly decoded in embedded struct.
-	TLSSetting `mapstructure:",squash"`
+	Config `mapstructure:",squash"`
```

Comment:
I'm not convinced that having a field named `Config` in a struct named `ClientConfig` is a good choice of names, but I cannot think of a better name. I don't think `CommonConfig` is any better? This field should ideally not exist at all, but well, Go lang... :roll_eyes:

Reply:
I played around with `TLSConfig Config `mapstructure:",squash"`